TAP-2.md

```diff
@@ -37,123 +37,84 @@ Provide all on-chain liquidity in a familiar limit orderbook interface to read (
 ### Nice-to-have Requirements
 - **Local fine-tuning**: Traders can request not only the provided points on the orderbook but also any arbitrary point in between. Those calls run a iterative search (solver) for the correct amount – within a certain window of tolerance.
 ## Not included
-# Implementation
+# Design (draft)
 *Just a draft for inspiration – take and leave what you like.*
-```ts
-// Core Domain Types
-type Side = 'buy' | 'sell'
-type GasMode = 'include' | 'exclude'
+## Core Components
+### 1. Protocol Stream Manager
+Manages connection to Tycho Indexer for real-time protocol state updates.
+```rust
+use tycho_core::dto::Chain;
+use tycho_client::stream::TychoStreamBuilder;
 
-type TokenPair = {
-  token0: string  // Base token address
-  token1: string  // Quote token address
-}
-
-type Route = {
-  protocols: string[]  // DEX protocols used
-  calldata: string    // Executable calldata including splits
-}
-
-type Level = {
-  price: string
-  size: string
-  route: Route
-  gasCost: {
-    wei: string
-    token0: string
-    token1: string
-  }
-}
-
-// Book State
-type OrderbookState = {
-  pair: TokenPair
-  timestamp: number
-  blockNumber: number
-  bids: Level[]  // Sorted best to worst
-  asks: Level[]  // Sorted best to worst
-}
-
-// Query Options
-type OrderbookOptions = {
-  maxHops?: number
-  maxSplits?: number
-  includedProtocols?: string[]
-  pricePoints?: number
-  gasMode: GasMode
-  gasToken?: string
-  maxGasPriceGwei?: string
-}
-
-type FinetuneOptions = OrderbookOptions & {
-  tolerance: string
-  maxIterations?: number
+pub struct ProtocolStreamManager {
+    tycho_url: String,
+    auth_key: Option<String>,
+    tvl_threshold: f64,
```

Comment:
Idk this seems a bit superfluous. I would the dev define this on as needed basis. Maybe mention in the requirements that this should use Tycho as it's main data source instead?

This seems like a very shallow interface and you could just use the `TychoStreamBuilder` directly instead.